File: MATEMÁTICA/1240/main.rb
# frozen_string_literal:true

def fits(b, a)
  num_digits_b = b.digits.length
  last_digits_a = a % 10**num_digits_b
  last_digits_a == b
end

n = gets.to_i
n.times do
  a, b = gets.split.map(&:to_i)
  if fits b, a
    puts 'encaixa'
  else
    puts 'nao encaixa'
  end
end

File: MATEMÁTICA/1240/main.dart
import 'dart:io';
import 'dart:math';

bool fits(int b, int a) {
  int numDigitsB = b.toString().length;
  int lastDigitsA = a % pow(10, numDigitsB).toInt();
  return lastDigitsA == b;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);
  for (int times = 0; times < n; times++) {
    var [a, b] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    if (fits(b, a))
      print('encaixa');
    else
      print('nao encaixa');
  }
}
